File: library/src/main/java/com/raizlabs/android/dbflow/sql/language/WhereBase.java
package com.raizlabs.android.dbflow.sql.language;

import com.raizlabs.android.dbflow.sql.Query;
import com.raizlabs.android.dbflow.structure.Model;

/**
 * Description: The base for a {@link Where} statement.
 */
public interface WhereBase<ModelClass extends Model> extends Query {

    /**
     * @return The table of this query.
     */
    Class<ModelClass> getTable();

    /**
     * @return The base Query object.
     */
    Query getQueryBuilderBase();
}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/query/WhereBase.kt
package com.raizlabs.android.dbflow.query

import com.raizlabs.android.dbflow.sql.Query
import com.raizlabs.android.dbflow.database.DatabaseWrapper

/**
 * Description: The base for a [Where] statement.
 */
interface WhereBase<TModel> : Query, Actionable {

    /**
     * @return The table of this query.
     */
    val table: Class<TModel>

    /**
     * @return The base Query object.
     */
    val queryBuilderBase: Query

    val databaseWrapper: DatabaseWrapper

}
